ci: Complete CI workflow

Kappuccino111 opened the pull request:
Completes and closes #15

Kappuccino111 commented:
@jonas-rem CodeSpell reports a lot of typos ( See the last check - Run CodeSpell ).Do you have anything in mind for the same ? 

I have managed to finish a base line implementation for both Java Linting and Codespell.

jonas-rem commented:
@Kappuccino111, you could check only the diff instead of the whole repository as is done in most existing checks. You could directly integrate it into the `check_compliance.py` script. That way we can also easily run the checks offline before pushing to the repo.

Kappuccino111 commented:
@jonas-rem pushed in 2 changes 

- [x] Removed testing on entire repository and removed code from compliance.yml
- [x] Added integration in check_compliance.py

Kappuccino111 reviewed (commented):
- [x] Added a `run_command()` to handle command execution and capture both stdout and stderr.
- [x] New Classes `CheckstyleTest()`and `CodespellTest()` inheriting from ComplianceTest base class.

jonas-rem commented on utils/ci/check_compliance.py:
```diff
@@ -501,6 +508,43 @@ def run(self):
+            print(stdout)
+            if returncode != 0:
+                self.failure(f"Checkstyle errors:\n{stderr}")
+
```
Add additional newline in-between classes (2).

jonas-rem reviewed (changes requested):
Regarding `codespell`: I tried testing by making a commit that adds misspelled words like `teh`, `recieve` but codespell still ran the check successful. How did you tested?

Regarding `checkstyle`: I get the following error:
```
ERROR   : Test Checkstyle failed: 
Checkstyle errors:
Error: Invalid or corrupt jarfile checkstyle-10.8.0-all.jar
compliance: exit 1 (1.90 seconds) /home/jonas/git/zephyrproject/lwm2m_server/utils/ci> python /home/jonas/git/zephyrproject/lwm2m_server/utils/ci/check_compliance.py pid=341301
  compliance: FAIL code 1 (1.93=setup[0.03]+cmd[1.90] seconds)
  evaluation failed :( (1.97 seconds)
```

Can we limit this PR to codespell and look into the java checks after that? I saw that Leshan offers a whole suite of checks that can all be executed with maven, which we already use to build our Leshan server implementation. This would cover much more points, like verifying the java build..

Kappuccino111 commented:
Deleted the original CI-workflow branch , will update with requested changes soon.
